client/components/search/index.jsx

```diff
@@ -271,19 +280,26 @@ const Search = React.createClass( {
 			spellCheck: 'false'
 		};
 
-		searchClass = classNames( this.props.additionalClasses, {
+		const searchClass = classNames( this.props.additionalClasses, this.props.dir, {
 			'is-expanded-to-container': this.props.fitsContainer,
 			'is-open': isOpenUnpinnedOrQueried,
 			'is-searching': this.props.searching,
-			rtl: this.props.dir === 'rtl',
-			ltr: this.props.dir === 'ltr',
+			'no-close-button': this.props.hideClose,
+			'has-focus': this.state.hasFocus,
 			search: true
 		} );
 
+		const gridIconClass = classNames( 'search-open__icon', this.props.dir );
+		const fadeDivClass = classNames( 'search__input-fade', this.props.dir );
+		const inputClass = classNames( 'search__input', this.props.dir, {
+			'no-close-button': this.props.hideClose
+		} );
+
 		return (
 			<div className={ searchClass } role="search">
 				<Spinner />
 				<div
+					className="search__component-icon-div"
```

Comment:
Yes, not a lot of meaning in `-div`. Taking into account function of this div think that the name `search__icon-navigation` makes sense.